src/components/utils/rqb-inputs/custom-react-query-builder.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import Grid from '@mui/material/Grid';
+import { QueryBuilderMaterial } from '@react-querybuilder/material';
+import {
+    Field,
+    formatQuery,
+    OptionList,
+    QueryBuilder,
+    RuleGroupTypeAny,
+} from 'react-querybuilder';
+import {
+    getOperators,
+    queryValidator,
+} from '../../dialogs/filter/expert/expert-filter-utils';
+import { ErrorInput, FieldErrorAlert } from '@gridsuite/commons-ui';
+import { useIntl } from 'react-intl';
+import { combinatorType } from '../../dialogs/filter/expert/expert-filter-constants';
+import { useFormContext } from 'react-hook-form';
+import RemoveButton from 'components/utils/rqb-inputs/remove-button';
+import CombinatorSelector from 'components/utils/rqb-inputs/combinator-selector';
+import AddButton from 'components/utils/rqb-inputs/add-button';
+import ValueEditor from './value-editor';
+import { EXPERT_FILTER_QUERY } from '../../dialogs/filter/expert/expert-filter-form';
+
+interface ExpertFilterRQBProps {
+    name: string;
+    fields: Field[];
+}
+
+const CustomReactQueryBuilder = (props: ExpertFilterRQBProps) => {
+    const { getValues, setValue, watch } = useFormContext();
+    const intl = useIntl();
+
+    const query = watch(props.name);
```

Comment:
May be useWatch is better to avoid re-render at root level, to verify, see https://junaidklair.medium.com/guide-to-react-form-hook-and-watch-vs-usewatch-962f3c30c6f4

Reply:
This is inspired from the CustomAgGridTable component, I changed it to useWatch() in this case as it seems superior for this case

Reply:
I changed it back to watch as otherwise there are issues with form validation. This is the recommended approach when dealing with external components: https://github.com/react-hook-form/react-hook-form/issues/456

Reply:
OK, we will discuss later..